### DISCUSSION TITLE
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)

### DISCUSSION THREAD

**Discussion by adrien-jagora** [taxonomicallyinformedannotation/tima#165] [Q&A]
Hi,
I tried to use timaR a few times this last month and I always ended up with the same error message during the sanitization of the spectra: 
Warning: Error in : Error running targets::tar_make()
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
Debugging guide: https://books.ropensci.org/targets/debugging.html
How to ask for help: https://books.ropensci.org/targets/help.html
Last error message:
    error in evaluating the argument 'object' in selecting a method for function 'addProcessing': error in evaluating the argument 'object' in selecting a method for function 'addProcessing': error in evaluating the argument 'object' in selecting a method for function 'filterIntensity': error in evaluating the argument 'object' in selecting a method for function 'filterIntensity': error in evaluating the argument 'object' in selecting a method for function 'filterFourierTransformArtefacts': error in evaluating the argument 'object' in selecting a method for function 'dropNaSpectraVariables': error in evaluating the argument 'object' in selecting a method for function 'Spectra': error reading from connection
Last error tr [... truncated]
  1: shiny::runApp
The version used was the 2.9.6 as for today but I am pretty sure I encountered the same problem with the 2.9.5. The only modification made to the base parameters was to select the "summarize results to one row per feature" option. Otherwise, a .mgf, a .csv and a .tsv were imported, but no Sirius job.
Where does the error come from and how can I resolve it ?
Thank you in advance !

**Comment by Adafede** [comment]
Hi @adrien-jagora,

I just added some new features to the last version I planned releasing later in the afternoon if all tests go well! This will be `2.9.7` then.

In the meantime, could you try removing any leftover from previous versions? Both in `~/.tima` and delete the repository you cloned and clone it freshly? This should ensure you got the latest version without eventual residuals.

Sorry for the inconvenience, but the new version should hopefully be worth it 😉

**Reply by Adafede** [reply]
Oh and please, do not wait a whole month before opening an issue/discussion! 😊

**Reply by adrien-jagora** [reply]
Thank you very much for your answer, @Adafede, I can't wait to test this new version !

**Reply by Adafede** [reply]
@adrien-jagora The checks I wanted are passing. Could you try the latest version and tell me if everything working as expected? If so, I'll release it tomorrow.

**Reply by adrien-jagora** [reply]
I just got this new error:
Listening on http://127.0.0.1:3838
2024-07-25 09:58:20.252815 Loading default params
2024-07-25 09:58:20.310095 Changing parameters ...
2024-07-25 09:58:20.312278 ... Small
2024-07-25 09:58:20.31878 ... Advanced
Registered S3 methods overwritten by 'callr':
  method                    from
  format.callr_status_error
  print.callr_status_error
• url: 'http://127.0.0.1:3839'
• host: '127.0.0.1'
• port: '3839'
tar_source(): these files do not exist: R
▶ dispatched target lib_spe_exp_int_pre
✖ errored target lib_spe_exp_int_pre
✖ errored pipeline [1.27 seconds]
Warning: Error in : Error running targets::tar_make()
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
Debugging guide: https://books.ropensci.org/targets/debugging.html
How to ask for help: https://books.ropensci.org/targets/help.html
Last error message:
    invalid 'file' argument
Last error traceback:
    prepare_libraries_spectra()
    file.exists(input)
    .handleSimpleError(function (condition)  {     state$error <- build_mess...
    h(simpleError(msg, call))
  1: shiny::runApp

**Reply by Adafede** [reply]
@adrien-jagora 
Sorry for that. Should be fixed now!

By the way, there is a new parameter in the options "high confidence only" that is ticked by default. It will remove tons of annotations you had previously. Do not forget to untick it in case 😉

**Reply by adrien-jagora** [reply]
Thank you @Adafede, I will test it right away ! And thank your for the tip also.

**Reply by adrien-jagora** [reply]
It looks like it is working ! So far, It is performing spectral comparison

**Reply by adrien-jagora** [reply]
Unfortunately, I got this new error in the end:
2024-07-25 13:15:14.809179 Applying sanitization of the spectra
✖ errored target ann_spe_pos
✖ errored pipeline [10.054 minutes]
Warning messages:
1: NAs introduced by coercion
2: NAs introduced by coercion
3: NAs introduced by coercion
4: incomplete final line found on 'data/source/libraries/sop/ecmdb.json.zip:ecmdb.json'
5: Invalid .internal.selfref detected and fixed by taking a (shallow) copy of the data.table so that := can add this new column by reference. At an earlier point, this data.table has been copied by R (or was created manually using structure() or similar). Avoid names<- and attr<- which in R currently (and oddly) may copy the whole data.table. Use set* syntax instead to avoid copying: ?set, ?setnames and ?setattr. If this message doesn't help, please report your use case to the data.table issue tracker so the root cause can be fixed or this message improved.
6: 3 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages.
Warning: Error in : Error running targets::tar_make()
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
Debugging guide: https://books.ropensci.org/targets/debugging.html
How to ask for help: https://books.ropensci.org/targets/help.html
Last error message:
    error in evaluating the argument 'object' in selecting a method for function 'addProcessing': error in evaluating the argument 'object' in selecting a method for function 'addProcessing': error in evaluating the argument 'object' in selecting a method for function 'filterIntensity': error in evaluating the argument 'object' in selecting a method for function 'filterIntensity': error in evaluating the argument 'object' in selecting a method for function 'filterFourierTransformArtefacts': error in evaluating the argument 'object' in selecting a method for function 'dropNaSpectraVariables': error in evaluating the argument 'object' in selecting a method for function 'Spectra': error reading from connection
Last error tr [... truncated]
  1: shiny::runApp

**Reply by Adafede** [reply]
Weird... it is actually the same as the first error...could it be that you have an old version of the `Spectra` package?

If you can try removing any previousl installed `Spectra`package version this might eventually help?

**Reply by adrien-jagora** [reply]
I have uninstalled R, RTools, RStudio and suppressed my tima-r folder, yet I still got the same error. Is there another way to make sure I suppressed everything related to an old version ?

**Reply by Adafede** [reply]
I meant uninstalling any previous version of timaR, R, RTools and Rstudio should not impact it.
If you can copy/paste the lines you obtain when running the `Rscript inst/scripts/install.R` command might eventually help

**Reply by adrien-jagora** [reply]
Those were the lines I obtained:
You should install RTools if not already done
You already have the latest version (2.9.7) skipping
Creating cache at C:/Users/adrie/.tima
Copying default architecture ...
Installing local targets
Getting local DESCRIPTION

**Reply by Adafede** [reply]
Yes, so it found a previous version of timaR you need to uninstall first

**Reply by adrien-jagora** [reply]
I suppressed everything, reinstalled it and still I have the same issue
